pkg/solana/txm/pendingtx.go

```diff
@@ -560,6 +579,162 @@ func (c *pendingTxContext) TrimFinalizedErroredTxs() int {
 	return len(expiredIDs)
 }
 
+func (c *pendingTxContext) GetSignatureInfo(sig solana.Signature) (txInfo, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	info, exists := c.sigToTxInfo[sig]
+	if !exists {
+		return txInfo{}, ErrSigDoesNotExist
+	}
+	return info, nil
+}
+
+func (c *pendingTxContext) OnReorg(id string) error {
+	err := c.withReadLock(func() error {
+		// Check if the transaction is still in a non finalized/errored state
+		var broadcastedExists, confirmedExists bool
+		_, broadcastedExists = c.broadcastedProcessedTxs[id]
+		_, confirmedExists = c.confirmedTxs[id]
+		if !broadcastedExists && !confirmedExists {
+			return ErrTransactionNotFound
+		}
+		return nil
+	})
+	if err != nil {
+		// If transaction or sig are not found, return
+		return err
+	}
+
+	var pTx pendingTx
```

Comment:
nit: think we can move this into the write lock block since we don't really need it outside